src/platform/common/stm32/pwm_output_dshot_shared.c

```diff
@@ -257,9 +276,11 @@ FAST_CODE_NOINLINE bool pwmTelemetryDecode(void)
 
     dshotTelemetryState.rawValueState = DSHOT_RAW_VALUE_STATE_NOT_PROCESSED;
     inputStampUs = 0;
-    dshotEnableChannels(dshotPwmDevice.count);
+    dshotEnableChannels(motorCount);
+
+
+#endif // USE_DSHOT_TELEMETRY
```

Comment:
MAybe reverse the `#ifdef USE_DSHOT_TELEMETRY` - this else branch is very far from test.

```
#ifndef USE_DSHOT_TELEMETRY
    return true;
#else
 ...

Reply:
With the inversion and return - no need for the else. Good pick up.